armotypes/pagination_structs.go

```diff
@@ -12,27 +12,29 @@ type RespTotal struct {
 }
 
 const (
-	V2ListExistsOperator   string = "exists"
-	V2ListEqualOperator    string = "equal"
-	V2ListMissingOperator  string = "missing"
-	V2ListEmptyOperator    string = "empty"
-	V2ListMatchOperator    string = "match"
-	V2ListGreaterOperator  string = "greater"
-	V2ListLowerOperator    string = "lower"
-	V2ListRegexOperator    string = "regex"
-	V2ListLikeOperator     string = "like"
-	V2ListRangeOperator    string = "range"
-	V2ListIgnoreCaseOption string = "ignorecase"
-	V2ListArrayOperator    string = "arraymatch"
+	V2ListExistsOperator       string = "exists"
+	V2ListEqualOperator        string = "equal"
+	V2ListMissingOperator      string = "missing"
+	V2ListEmptyOperator        string = "empty"
+	V2ListMatchOperator        string = "match"
+	V2ListGreaterOperator      string = "greater"
+	V2ListLowerOperator        string = "lower"
+	V2ListRegexOperator        string = "regex"
+	V2ListLikeOperator         string = "like"
+	V2ListRangeOperator        string = "range"
+	V2ListIgnoreCaseOption     string = "ignorecase"
+	V2ListElementMatchOperator string = "elemMatch" // for matching elements in array e.g. {"myObjectsArray": "name:John;age:30|elemMatch"}
```

Comment:
I think we need here something more powerfull.
Like add the elemMatch in the inner filter key, not in the value.
e.g. 
```json
{
  "myObjectsArray.name|elemMatch": "John",
  "myObjectsArray.age|elemMatch": "30"
}
```

Reply:
hmmm
It is nicer to read and compose 
How ever it is less powerful since you cannot create a one filter with match OR match
You cannot also create nested elemMatch queries inside elemMatch queries

Reply:
This was my thing, didn't elaborated here, but 
```json
{
  "myObjectsArray.name|elemMatch:0": "John",
  "myObjectsArray.age|elemMatch:0": "30",
  "myObjectsArray.name|elemMatch:1": "Due",
  "myObjectsArray.age|elemMatch:1": "55",
  "myObjectsArray|elemMatch:3|.name|elemMatch:4": "Due",
  "myObjectsArray|elemMatch:3|.age|elemMatch:4": "55"

}
```
If you need OR, you put it in another filters map.

Reply:
done